app/lib/middlewares/set_dev_client_geo.rb

```diff
@@ -0,0 +1,16 @@
+module Middlewares
+  class SetDevClientGeo
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      request = Rack::Request.new(env)
+      client_geo = request.params["client_geo"]
```

Comment:
Yes to all of this, and to expand on the last point: since we load some (and increasingly more) billboards asynchronously, this approach will need a companion bit of JavaScript to grab the param and attach it to the relevant XHR/fetch requests. Beyond that I think it's alright that the param isn't sticky, since it's only for dev purposes - my experience of testing out the location targeting was visiting feeds and articles directly